### PR TITLE
Footer new guidelines

### PR DESCRIPTION
**Pull request by nicl**
## What does this change?

Converts footer to new component guidelines.

## Why?

So that we are consistent and simple.

### CODE REVIEW

File: frontend/components/Footer/index.js [hunk 1]
```diff
@@ -110,87 +110,89 @@ const footerLinks: Array<Array<Link>> = [
...
+    display: flex;
+    flex-wrap: wrap;
+    flex-direction: row;
+    border-top: 1px solid #434343;
+
+    ${tablet} {
+        border-top: none;
+    }
+
+    ul {
```
Comment (nicl) [inline frontend/components/Footer/index.js]: Nesting 😬 but I think this is probably okay here as it's only one level and avoids having an extra, even more specifically named, css rule for the list in the div.

**Comment by SiAdcock** [comment]
Should we move `components/Footer/index.js` into `components/Footer.js`? Having a separate folder seems superfluous

File: frontend/components/Footer/index.js [hunk 2]
```diff
@@ -200,36 +202,38 @@ const FooterLinks = ({ links }: Props) => {
...
 
-    return <div className="footer-links">{linkGroups}</div>;
+    return <div className={footerList}>{linkGroups}</div>;
 };
 
 export default () => (
-    <Footer>
-        <Main className="footer-inner">
+    <footer className={footer}>
+        <Main className={footerInner}>
```
Comment (SiAdcock) [inline frontend/components/Footer/index.js]: There should only be one `<main>` element per page, and it shouldn't be a descendent of a `<footer>`

http://w3c.github.io/html/grouping-content.html#the-main-element

File: frontend/components/Footer/index.js (hunk 1 above)
Comment (SiAdcock) [reply frontend/components/Footer/index.js]: I think it's useful because it keeps the styles for the flex parent and flex children together.

**Comment by SiAdcock** [comment]
I've left some comments unrelated to your changes, but I think the way the styles and components are defined is 👌👌👌. To me, it's so much more readable, easier to follow and easier to jump from JSX to relevant CSS. The CSS looks like CSS, which is great.

**Comment by nicl** [comment]
@SiAdcock thanks. I've updated based on the feedback. But agree, the new approach feels a lot simpler to read and write :)

**Comment by AWare** [comment]
I'm not sure that this should be shared as css instead of making a component. You'll have to use `props.children` though, and I'm not sure what the right Flow typing would be.

**Comment by nicl** [comment]
@AWare hey thanks. Why do you think it's more useful as props here?

**Comment by AWare** [comment]
I don't know, the way it's being used feels more like a component. 

(re children and types, I think I'm happy with the pattern here https://github.com/guardian/dotcom-rendering/pull/132/files#diff-354d667a67b7715bee6fe379b3934bb6)

**Comment by nicl** [comment]
@SiAdcock re 'Maybe container belongs in @guardian/pasteup? It feels more like a sub-component-sized token than a component in itself.'

Something I'm not sure of is how to share css like this from pasteup, as we want to avoid the react-emotion dependency. We could export tagged template literals without wrapping them in css. It doesn't feel ideal for the client, as still need to wrap in css, but not too bad. I'm implemented that anyway to illustrate (but we can always pull the commit).

**Comment by SiAdcock** [comment]
Yes, the CSS duplication is not ideal. I think I'm more swayed by @AWare and his `<Container>` component solution, for the reasons I [ruminated about above](https://github.com/guardian/dotcom-rendering/pull/124#issuecomment-412869617)

**Comment by nicl** [comment]
@SiAdcock @AWare thanks both for thoughts. I've created a 'Container' component in Guui. I've put some thoughts in the commit message but basically - I think this is fine, but I have a preference for sharing css output, rather than components for purely visual stuff. But it's not a very strong preference!

**Comment by SiAdcock** [comment]
@nicl I think I generally agree with that sentiment. Let's see how this plays out during the simple article development, we can always switch to the `css` approach if the component approach starts to trip us up.

**Pull request merged**